hip/stop/residual_norm_kernels.hip.cpp

```diff
@@ -125,6 +125,86 @@ GKO_INSTANTIATE_FOR_EACH_NON_COMPLEX_VALUE_TYPE(
 
 
 }  // namespace residual_norm
+
+
+/**
+ * @brief The Residual norm stopping criterion namespace.
+ * @ref resnorm
+ * @ingroup resnorm
+ */
+namespace implicit_residual_norm {
+
+
+constexpr int default_block_size = 512;
+
+
+template <typename ValueType>
+__global__
+    __launch_bounds__(default_block_size) void implicit_residual_norm_kernel(
+        size_type num_cols, ValueType rel_residual_goal,
+        const ValueType *__restrict__ tau,
+        const ValueType *__restrict__ orig_tau, uint8 stoppingId,
+        bool setFinalized, stopping_status *__restrict__ stop_status,
+        bool *__restrict__ device_storage)
+{
+    const auto tidx = thread::get_thread_id_flat();
+    if (tidx < num_cols) {
+        if (sqrt(tau[tidx]) < rel_residual_goal * orig_tau[tidx]) {
+            stop_status[tidx].converge(stoppingId, setFinalized);
+            device_storage[1] = true;
```

Comment:
If two RHS converge at the same time, there would a write conflict here, no? Or does it not matter since all threads are writing the same value, similar to below? Maybe use an atomic write here anyway - this is not performance-sensitive at all.

Reply:
I think atomic is not necessary here for the same reason as mentioned below. I think adding an atomic write here might be a bit more confusing and I feel as it is now is a bit more clear.

Reply:
Only cuda-memcheck/ThreadSanitizer complaining, but that should not be an issue.